File: 0009-palindrome-number/0009-palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    let reverseOfX = 0
    let num = x
    if(x < 0)
        return false
    if(x < 10)
        return true
    while(num != 0) {

        reverseOfX = reverseOfX * 10 + Math.floor(num % 10)
        num = Math.floor(num / 10)
    }
    
    console.log(reverseOfX)
    
    return reverseOfX == x
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int num = x;
        int revNum = 0;
        
        // reverse the number
        if(num < 0)
            return false;
        while(num != 0) {
            revNum = (revNum *10) + (num % 10) ;
            num = num / 10;
        }
        
        
        // check if the two numbers are equal
        if(x == revNum)
            return true;
        return false;
    }
}